phoenix-core/src/it/java/org/apache/phoenix/iterate/PhoenixQueryTimeoutIT.java

```diff
@@ -202,6 +204,57 @@ public void testScanningResultIteratorQueryTimeoutForPagingWithVeryLowTimeout()
         }
     }
 
+    @Test
+    public void testScanningResultIteratorShouldNotQueryTimeoutForPagingAfterReceivingAValidRow() throws Exception {
+        //Arrange
+        PreparedStatement ps = loadDataAndPreparePagedQuery(10,1);
+
+        ManualEnvironmentEdge injectEdge;
+        injectEdge = new ManualEnvironmentEdge();
+        injectEdge.setValue(EnvironmentEdgeManager.currentTimeMillis());
+        EnvironmentEdgeManager.injectEdge(injectEdge);
+
+        //Act + Assert
```

Comment:
Please add more readable comment here. For example, this query should not timeout as the first row will be retrieved within 10 ms.